x-pack/plugins/lists/server/services/exception_lists/get_exception_list_summary.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  ExceptionListSummarySchema,
+  IdOrUndefined,
+  ListIdOrUndefined,
+  NamespaceType,
+} from '@kbn/securitysolution-io-ts-list-types';
+import { getSavedObjectType } from '@kbn/securitysolution-list-utils';
+
+import {
+  SavedObjectsClientContract,
+  SavedObjectsErrorHelpers,
+} from '../../../../../../src/core/server/';
+import { ExceptionListSoSchema } from '../../schemas/saved_objects';
+
+interface GetExceptionListSummaryOptions {
+  id: IdOrUndefined;
+  listId: ListIdOrUndefined;
+  savedObjectsClient: SavedObjectsClientContract;
+  namespaceType: NamespaceType;
+}
+
+interface ByOsAggBucketType {
+  key: string;
+  doc_count: number;
+}
+interface ByOsAggType {
+  by_os: {
+    buckets: ByOsAggBucketType[];
+  };
+}
+
+export const getExceptionListSummary = async ({
+  id,
+  listId,
+  savedObjectsClient,
+  namespaceType,
+}: GetExceptionListSummaryOptions): Promise<ExceptionListSummarySchema | null> => {
+  const savedObjectType = getSavedObjectType({ namespaceType });
+  let finalListId: string = listId ?? '';
+
+  // If id and no listId, get the list by id to use the list_id for the find below
+  if (listId === null && id != null) {
+    try {
+      const savedObject = await savedObjectsClient.get<ExceptionListSoSchema>(savedObjectType, id);
+      finalListId = savedObject.attributes.list_id;
+    } catch (err) {
+      if (SavedObjectsErrorHelpers.isNotFoundError(err)) {
+        return null;
+      } else {
+        throw err;
+      }
+    }
+  }
+
+  const savedObject = await savedObjectsClient.find<ExceptionListSoSchema>({
+    aggs: {
+      by_os: {
+        terms: {
+          field: `${savedObjectType}.attributes.os_types`,
+        },
+      },
+    },
+    filter: `${savedObjectType}.attributes.list_type: item`,
+    perPage: 0,
+    search: finalListId,
+    searchFields: ['list_id'],
+    sortField: 'tie_breaker_id',
+    sortOrder: 'desc',
+    type: savedObjectType,
+  });
+
+  const summary: ExceptionListSummarySchema = (savedObject.aggregations as ByOsAggType).by_os.buckets.reduce(
```

Comment:
Small tip: `savedObjectsClient.find` accepts aggregations type as the second type argument. You can save an extra type casting by providing it:

```suggestion
  const savedObject = await savedObjectsClient.find<ExceptionListSoSchema, ByOsAggType>({
    aggs: {
      by_os: {
        terms: {
          field: `${savedObjectType}.attributes.os_types`,
        },
      },
    },
    filter: `${savedObjectType}.attributes.list_type: item`,
    perPage: 0,
    search: finalListId,
    searchFields: ['list_id'],
    sortField: 'tie_breaker_id',
    sortOrder: 'desc',
    type: savedObjectType,
  });

  const summary: ExceptionListSummarySchema = savedObject.aggregations?.by_os.buckets.reduce(
```

Reply:
Wow!! Thank you so much!! Super helpful!